mmv1/products/cloudquotas/QuotaPreference.yaml

```diff
@@ -0,0 +1,164 @@
+    description: |
+      If set to true, and the quota preference is not found, a new one will be created. In this situation, updateMask is ignored.
+  - !ruby/object:Api::Type::Boolean
+    name: 'validate_only'
+    url_param_only: true
+    description: |
+      If set to true, validate the request, but do not actually update. Note that a request being valid does not mean that the request is guaranteed to be fulfilled.
+  - !ruby/object:Api::Type::Enum
+    name: 'ignore_safety_checks'
+    url_param_only: true
+    default_value: :QUOTA_SAFETY_CHECK_UNSPECIFIED
+    description: The list of quota safety checks to be ignored.
+    values:
+      - :QUOTA_SAFETY_CHECK_UNSPECIFIED
+      - :QUOTA_DECREASE_BELOW_USAGE
+      - :QUOTA_DECREASE_PERCENTAGE_TOO_HIGH
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The resource name of the quota preference. Required except in the CREATE requests.
+  - !ruby/object:Api::Type::String
+    name: 'service'
+    required: true
+    description: The name of the service to which the quota preference is applied.
+  - !ruby/object:Api::Type::String
+    name: 'quotaId'
+    required: true
+    description: |
+      The id of the quota to which the quota preference is applied. A quota id is unique in the service.
+      Example: CPUS-per-project-region.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'quotaConfig'
+    required: true
+    description: The preferred quota configuration.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'preferredValue'
+        required: true
+        description: |
+          The preferred value. Must be greater than or equal to -1. If set to -1, it means the value is "unlimited".
+      - !ruby/object:Api::Type::String
+        name: 'stateDetail'
+        output: true
+        description: Optional details about the state of this quota preference.
+      - !ruby/object:Api::Type::String
+        name: 'grantedValue'
+        output: true
+        description: Granted quota value.
+      - !ruby/object:Api::Type::String
+        name: 'traceId'
+        output: true
+        description: |
+          The trace id that the Google Cloud uses to provision the requested quota. This trace id may be used by the client to contact Cloud support to track the state of a quota preference request. The trace id is only produced for increase requests and is unique for each request. The quota decrease requests do not have a trace id.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'annotations'
+        description: |-
+          The annotations map for clients to store small amounts of arbitrary data. Do not put PII or other sensitive information here. See https://google.aip.dev/128#annotations.
+
+          An object containing a list of "key: value" pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+      - !ruby/object:Api::Type::Enum
+        name: 'requestOrigin'
```

Comment:
Since this is output-only, it is usually simpler to define it as a string. Then we don't need to manage a client-side set of options.

Reply:
I see, switched to string.

Reply:
Done, switched to string.